app/Services/IncidentTypeService.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace App\Services;
+
+use App\IncidentType;
+use Dingo\Api\Exception\StoreResourceFailedException;
+use Illuminate\Support\Facades\Validator;
+
+class IncidentTypeService
+{
+    /**
+     * @param array $data
+     * @return IncidentType
+     * @throws @StoreResourceFailedException
+     * @throws @\Exception
+     */
+    public function create(array $data): IncidentType
+    {
+        $this->validate($data);
+
+        $incidentType = new IncidentType();
+        $incidentType = $this->hydrate($data, $incidentType);
+        $incidentType->save();
+
+        return $incidentType;
+    }
+
+    /**
+     * @param IncidentType $incidentType
+     * @param array $data
+     * @return IncidentType
+     */
+    public function update(IncidentType $incidentType, array $data)
+    {
+        $this->validate($data, $incidentType->id);
+        $incidentType = $this->hydrate($data, $incidentType);
+        $incidentType->save();
+
+        return $incidentType;
+    }
+
+    /**
+     * @param array $data
+     * @param null $id
+     */
+    public function validate(array $data, $id = null)
```

Comment:
Was there a reason you added the id as part of the method signature? I don't see it used anywhere.

Reply:
It is used in $rules, 'code' => 'required|unique:incident_types,code,' . $id, more exactly when you want to update an existing object you will get a unique constraint error if you don't pass the current id.